test/e2e/e2e-upgrade.test.ts

```diff
@@ -0,0 +1,167 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+// tslint:disable: no-console
+
+import { expect } from '@oclif/test'
+import * as execa from 'execa'
+
+import { CheGithubClient } from '../../src/api/github-client'
+import { KubeHelper } from '../../src/api/kube'
+import { isKubernetesPlatformFamily } from '../../src/util'
+
+import { DEVFILE_URL, E2eHelper } from './util'
+
+const kube = new KubeHelper()
+const helper = new E2eHelper()
+jest.setTimeout(1000000)
+
+const binChectl = `${process.cwd()}/bin/run`
+
+const PLATFORM = process.env.PLATFORM || 'minikube'
+
+const INSTALLER = 'operator'
+const NAMESPACE = 'eclipse-che'
+
+const NIGHTLY = 'nightly'
+
+const UPDATE_CHE_TIMEOUT_MS = 5 * 60 * 1000
+const WORKSPACE_START_TIMEOUT_MS = 5 * 60 * 1000
+const CHE_VERSION_TIMEOUT_MS = 50 * 1000
+
+async function runCommand(command: string, args?: string[]): Promise<string> {
+  console.log(`Running command: ${command} ${args ? args.join(' ') : ''}`)
+  const { exitCode, stdout, stderr } = await execa(command, args, { shell: true })
+  console.log(stdout)
+  if (exitCode !== 0) {
+    console.log(stderr)
+  }
+
+  expect(exitCode).equal(0)
+
+  return stdout
+}
+
+async function waitForVersionInCheCR(version: string, timeoutMs: number): Promise<boolean> {
```

Comment:
Usually wait methods return void or throw an error.
Should be moved to utility class.

Reply:
ok